### PR TITLE
CORE-7854 Backchain resolution

### PR DESCRIPTION
**Pull request by lankydan**

### CODE REVIEW

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt [hunk 1]
```diff
@@ -93,4 +93,13 @@ interface UtxoLedgerService {
...
     // TODO CORE-7327 Add verify(signedTx) verify(ledgerTx)
+
+    @Suspendable
+    fun persistTransaction(signedTransaction: UtxoSignedTransaction)
+
+    @Suspendable
+    fun resolveBackchain(signedTransaction: UtxoSignedTransaction, session: FlowSession)
+
+    @Suspendable
+    fun sendBackchain(session: FlowSession)
```
Comment (blsemo) [inline ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt]: An interesting question whether these need to be on the public API (in the long run) - shouldn't all this happen under the hood when you verify a transaction?
But probably ok for now to get it all moving...

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt [hunk 2]
```diff
@@ -68,13 +69,15 @@ interface UtxoTransactionBuilder {
...
-    fun addInputState(stateAndRef: StateAndRef<*>): UtxoTransactionBuilder
+    fun addInputState(stateRef: StateRef): UtxoTransactionBuilder
+
+//    /**
+//     * Adds an input state to the current [UtxoTransactionBuilder].
+//     *
+//     * @param stateAndRef The [StateAndRef] of the input state to add to the current [UtxoTransactionBuilder].
+//     * @return Returns a [UtxoTransactionBuilder] including the additional input state.
+//     */
+//    fun addInputState(stateAndRef: StateAndRef<*>): UtxoTransactionBuilder
```
Comment (blsemo) [inline ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt]: What is the reasoning behind this change? And why only for inputs, not for reference inputs?

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt (hunk 1 above)
Comment (lankydan) [reply ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt]: Yes, these are 100% temporary and don't see them being needed on the API in the future.

**Review by blsemo — APPROVED** [review]
LGTM

**Pull request merged**